### ISSUE TITLE
Some MODIS Terra visible band images incorrect in low reflectance regions

### ISSUE THREAD

**Issue by kathys** [davidh-ssec/polar2grid#126]
I noticed that there are some Terra MODIS bands gtiff files where  very low reflectance regions, like water bodies, are showing up as white.  These look much different than when created using a previous version of p2g.  

As an example, I placed all of the terra input data files, the output gtiff files and the ones I got from the p2g version that Liam is using on this ftp site:  ftp://ftp.ssec.wisc.edu/pub/ssec/kathys/modis_test/

In particular, take a look at visible band 7 images.  I also include the log file that I used to create the output Terra images (modis2gtiff.sh -f ../l1b/t1*.hdf).  Did I include enough info?  Just let me know if you need more.

Here is what I am talking about:  
![terra_modis_vis07_20170104_173200_wgs84_fit](https://cloud.githubusercontent.com/assets/1848019/21733153/97b769ae-d422-11e6-9dec-d69c5d0b1ea8.jpeg)

**Comment by davidh-ssec** [comment]
I wonder if this is a new case of fill values in the file not being caught properly. I believe band 7 was added after a user requested it, but no additional info was given ("oh it should be simple to add") and it may have some "hidden" fill values like band 2. I'll take a look.

**Comment by davidh-ssec** [comment]
@kathys I figured it out. When provided the 500m file it will use that to get band07 (and all other bands it can get from that file). This file has an undocumented fill value 65530 in the "odd" sections of the image. Any idea what this value means?

If you only provide the 1km file you get the expected image. If this is a special fill value I can filter it. I could also use `valid_range` defined in the file to mask the data, but I think I had removed it previously since we were using the "saturated" and "can't aggregate" fill values specially and setting them to max brightness. I could still make this work, but we'd have to make sure it doesn't change anything in the final output compared to previous versions.

So put simply, the 500m image is wrong, the 1km is still correct.

**Comment by kathys** [comment]
Here is a screen capture from a document that describes some of the 655xx values.  Anything in this range means there is a problem.  This is taken from the MODIS Level 1B Product User’s Guide.  
![screen shot 2017-01-09 at 10 25 09 am](https://cloud.githubusercontent.com/assets/1848019/21773801/088089c4-d656-11e6-81d6-666edfc22c44.png)
